reservation_units/tasks.py

```diff
@@ -14,3 +21,41 @@ def _update_reservation_unit_pricings() -> None:
     logger.info(f"Updating reservation unit pricing with date {today}")
     num_updated = update_reservation_unit_pricings()
     logger.info(f"Updated {num_updated} reservation units with date {today}")
+
+
+@app.task(
+    name="refresh_reservation_unit_product_mapping",
+    autoretry_for=(TypeError,),
+    max_retries=5,
+    retry_backoff=True,
+)
+def refresh_reservation_unit_product_mapping(reservation_unit_pk) -> None:
+    from reservation_units.models import ReservationUnit
+
+    reservation_unit = ReservationUnit.objects.filter(pk=reservation_unit_pk).first()
+    if reservation_unit is None:
+        capture_message(
+            f"Unable to refresh reservation unit product mapping. Reservation unit not found: {reservation_unit_pk}",
+            level="warning",
+        )
+        return
+
+    payment_merchant = ReservationUnitPaymentHelper.get_merchant(reservation_unit)
+
+    if ReservationUnitPaymentHelper.requires_product_mapping_update(reservation_unit):
+        params = CreateProductParams(
+            namespace=settings.VERKKOKAUPPA_MERCHANT_NAMESPACE,
+            namespace_entity_id=reservation_unit.pk,
+            merchant_id=payment_merchant.id,
+        )
+        api_product = create_product(params)
+        payment_product, _ = PaymentProduct.objects.update_or_create(
+            id=api_product.product_id, defaults={"merchant": payment_merchant}
+        )
+        reservation_unit.payment_product = payment_product
+
+    # Remove product mapping if merchant is removed
+    if reservation_unit.payment_product is not None and payment_merchant is None:
+        reservation_unit.payment_product = None
+
+    reservation_unit.save(run_product_refresh=False)
```

Comment:
`ReservationUnit.objects.filter(id=reservation_unit.id).update(payment_product=payment_product)`
Wouldn't call instance's `save()` method and then the `run_product_refresh` could be ditched => but either way is proper  I guess

Reply:
Awwwwyeessss! So much better 👏